cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java

```diff
@@ -719,8 +722,38 @@
     public static final String LOGIN_MODULE_CLASS_NAME = "security.authentication.loginmodule.className";
     /** Realm file for Basic Authentication */
     public static final String BASIC_REALM_FILE = "security.authentication.basic.realmfile";
-    /** Enables SSL */
+    /** Enables external SSL */
+    @Deprecated
     public static final String SSL_ENABLED = "ssl.enabled";
+    /** Key to mark a discoverable which supports ssl */
+    public static final String SSL_DISCOVERABLE_KEY = "ssl";
+
+    /**
+     * App Fabric
+     */
+    public static final class SSL {
+      /** Enables SSL for external services. */
+      @SuppressWarnings("unused")
+      public static final String EXTERNAL_ENABLED = "ssl.external.enabled";
+      /** Enables SSL for internal services. */
+      public static final String INTERNAL_ENABLED = "ssl.internal.enabled";
+      /** Password for the java keystore. */
+      public static final String KEYSTORE_PASSWORD = "ssl.internal.keystore.password";
+      /** Type for the java keystore. e.g. JCEKS. */
+      public static final String KEYSTORE_TYPE = "ssl.internal.keystore.type";
+      /** Algorithm used to generate key pair used in SSL certificate generation. */
+      public static final String KEY_PAIR_ALGORITHM = "ssl.internal.key.pair.algorithm";
+      /** Algorithm used for generating the random number used for SSL certificate. */
+      public static final String SECURE_RANDOM_ALGORITHM = "ssl.internal.random.algorithm";
+      /** Provider for the secure random number generator. */
+      public static final String SECURE_RANDOM_PROVIDER = "ssl.internal.random.provider";
+      /** Validity of the generated SSL certificate. */
+      public static final String CERT_VALIDITY = "ssl.internal.cert.validity";
+      /** Distinguished name to be used while generating the SSL certificate */
+      public static final String CERT_DISTINGUISHED_NAME = "ssl.internal.cert.distinguished.name";
+      /** Algorithm used for self signing the generated certificate. */
+      public static final String SIGNATURE_ALGORITHM = "ssl.internal.cert.sign.algorithm";
```

Comment:
It seems like we are having way too many unnecessary configurations for the generated cert. If the cert is generated, why all these need to be configurable? If someone try to control all these, why not provide a proper cert (if we support it, if not, still there is no point in configuring all these)?